### PR TITLE
feat(seo): add initial package scaffold

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## Purpose
This adds a package as described by #1632 

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: libs/seo/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# @daffodil/seo
+
+The `@daffodil/seo` library intends to provide:
+
+1. Platform-agnostic drivers for working with customer orders.
```
Comment (damienwebdev) [inline libs/seo/README.md:5]: docs.

File: libs/seo/package.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{
+  "name": "@daffodil/seo",
+  "version": "0.0.0-PLACEHOLDER",
+  "description": "A client-side Angular package for managing ecommerce orders agnostic to the ecommerce platform.",
```
Comment (damienwebdev) [inline libs/seo/package.json:4]: docs

File: libs/seo/package.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  },
+  "scripts": {
+    "build": "ng build seo --prod",
+    "lint": "cd ../.. && ng lint seo",
+    "lint:fix": "npm run lint -- --fix",
+    "test": "ng test seo --watch=false --browsers=ChromeHeadless",
+    "publish": "(cd ../../dist/seo; npm publish --access=public)"
+  },
+  "homepage": "https://github.com/graycoreio/daffodil",
+  "peerDependencies": {
```
Comment (damienwebdev) [inline libs/seo/package.json:22]: deps

File: libs/seo/package.json [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/graycoreio/daffodil/issues"
+  },
+  "scripts": {
+    "build": "ng build seo --prod",
+    "lint": "cd ../.. && ng lint seo",
+    "lint:fix": "npm run lint -- --fix",
+    "test": "ng test seo --watch=false --browsers=ChromeHeadless",
+    "publish": "(cd ../../dist/seo; npm publish --access=public)"
```
Comment (damienwebdev) [inline libs/seo/package.json:19]: don't publish.

File: libs/seo/tsconfig.lib.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    "target": "es2015",
+    "declaration": true,
+    "inlineSources": true,
+    "types": [],
+    "lib": [
+      "dom",
+      "es2018"
+    ]
+  },
+  "angularCompilerOptions": {
```
Comment (damienwebdev) [inline libs/seo/tsconfig.lib.json:14]: Can you compare these against a new ng12 package. We haven't reviewed these in a while, and it's worth it just to gloss over and compare.

**Review by damienwebdev — CHANGES_REQUESTED** [review]

File: libs/seo/package.json (hunk 4 above)
Comment (griest024) [reply libs/seo/package.json]: fixed

File: libs/seo/package.json (hunk 3 above)
Comment (griest024) [reply libs/seo/package.json:22]: fixed

File: libs/seo/package.json (hunk 2 above)
Comment (griest024) [reply libs/seo/package.json]: fixed

File: libs/seo/README.md (hunk 1 above)
Comment (griest024) [reply libs/seo/README.md]: fixed

File: libs/seo/tsconfig.lib.json (hunk 5 above)
Comment (griest024) [reply libs/seo/tsconfig.lib.json:14]: looking at the [schematic template](https://github.com/angular/angular-cli/blob/master/packages/schematics/angular/library/files/tsconfig.lib.json.template), they look fairly similar.

Here's what `ng g library` outputs (for our current ng version):
```json
{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "outDir": "../out-tsc/lib",
    "target": "es2015",
    "declaration": true,
    "declarationMap": true,
    "inlineSources": true,
    "types": [],
    "lib": [
      "dom",
      "es2018"
    ]
  },
  "angularCompilerOptions": {
    "skipTemplateCodegen": true,
    "strictMetadataEmit": true,
    "enableResourceInlining": true
  },
  "exclude": [
    "src/test.ts",
    "**/*.spec.ts"
  ]
}
```

File: libs/seo/tsconfig.lib.json [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    "annotateForClosureCompiler": true,
+    "skipTemplateCodegen": true,
+    "strictMetadataEmit": true,
+    "fullTemplateTypeCheck": true,
+    "strictInjectionParameters": true,
+    "enableResourceInlining": true,
+    "allowSyntheticDefaultImports": true
```
Comment (damienwebdev) [inline libs/seo/tsconfig.lib.json:21]: Let's add the strict templates key and set it to true.

File: libs/seo/tsconfig.lib.json [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    "declaration": true,
+    "inlineSources": true,
+    "types": [],
+    "lib": [
+      "dom",
+      "es2018"
+    ]
+  },
+  "angularCompilerOptions": {
+    "annotateForClosureCompiler": true,
```
Comment (damienwebdev) [inline libs/seo/tsconfig.lib.json:15]: I don't think we need this: https://github.com/angular/angular/issues/37234

**Review by damienwebdev — CHANGES_REQUESTED** [review]

File: libs/seo/tsconfig.lib.json (hunk 7 above)
Comment (griest024) [reply libs/seo/tsconfig.lib.json]: removed

File: libs/seo/tsconfig.lib.json (hunk 6 above)
Comment (griest024) [reply libs/seo/tsconfig.lib.json:21]: added

**Comment by damienwebdev** [comment]
Can you add an entry to the contribution guidelines?

**Comment by griest024** [comment]
> Can you add an entry to the contribution guidelines?

added

**Review by damienwebdev — APPROVED** [review]
lgtm

**Pull request merged**